src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp

```diff
@@ -688,19 +688,22 @@ void DistributedSearchTreeImpl<DeviceType>::filterResults(
       Kokkos::RangePolicy<ExecutionSpace>(0, n_queries), KOKKOS_LAMBDA(int q) {
         auto local_buffer = Kokkos::subview(
             buffer, Kokkos::make_pair(offset(q), offset(q + 1)));
-        PriorityQueue queue(UnmanagedStaticVector<PairIndexDistance>(
-            local_buffer.data(), local_buffer.size()));
-        for (int i = offset(q); i < offset(q + 1); ++i)
-          queue.emplace(Kokkos::Array<int, 2>{{indices(i), ranks(i)}},
-                        distances(i));
-
-        int count = 0;
-        while (!queue.empty() && count < queries(q)._k)
+        if (local_buffer.extent(0) > 0)
```

Comment:
Do this check *prior* constructing the Kokkos::subview, as that cost is not trivial.

Reply:
Done